src/main/java/life/qbic/App.java

```diff
@@ -1,159 +1,76 @@
-            System.out.println("Arguments --identifier and --file are mutually exclusive, please provide only one.");
-            System.exit(1);
-        } else if (commandLineParameters.filePath != null) {
-            commandLineParameters.ids = Argparser.readProvidedIdentifiers(commandLineParameters.filePath.toFile());
-        }
-
-        // Set the server URLS specified by a config file/CLI argument -> use default if none is provided
-        AS_URL = commandLineParameters.as_url;
-        DSS_URL = commandLineParameters.dss_url;
-
-        System.out.format("Please provide password for user \'%s\':\n", commandLineParameters.user);
-
-        String password = Argparser.readPasswordFromInputStream();
+  private static final Logger LOG = LogManager.getLogger(App.class);
 
-        if (password.isEmpty()) {
-            System.out.println("You need to provide a password.");
-            System.exit(1);
-        }
+  public static void main(String[] args) throws IOException {
+    // parse and verify all commandline parameters
+    PostmanCommandLineOptions commandLineParameters =
+        CommandLineParser.parseAndVerifyCommandLineParameters(args);
 
-        QbicDataLoader qbicDataLoader = new QbicDataLoader(AS_URL, DSS_URL, commandLineParameters.user, password,
-                commandLineParameters.bufferMultiplier * 1024, commandLineParameters.datasetType,
-            commandLineParameters.conservePath);
-        int returnCode = qbicDataLoader.login();
-        LOG.info(String.format("OpenBis login returned with %s", returnCode));
-        if (returnCode != 0) {
-            LOG.error("Connection to openBIS failed.");
-            System.exit(1);
-        }
-        LOG.info("Connection to openBIS was successful.");
+    // login to OpenBIS
+    QbicDataDownloader qbicDataDownloader = loginToOpenBIS(commandLineParameters);
 
-        LOG.info(String.format("%s provided openBIS identifiers have been found: %s",
-                commandLineParameters.ids.size(), commandLineParameters.ids.toString()));
+    // download all requested files by the user
+    qbicDataDownloader.downloadRequestedFilesOfDatasets(commandLineParameters, qbicDataDownloader);
+  }
 
-        // a suffix was provided -> only download files which contain the suffix string
-        if (!commandLineParameters.suffixes.isEmpty()) {
-            for (String ident : commandLineParameters.ids) {
-                LOG.info(String.format("Downloading files for provided identifier %s", ident));
-                List<IDataSetFileId> foundSuffixFilteredIDs = qbicDataLoader.findAllSuffixFilteredIDs(ident, commandLineParameters.suffixes);
+  /**
+   * logs into OpenBIS asks for and verifies password
+   *
+   * @param commandLineParameters
+   * @return
+   */
+  private static QbicDataDownloader loginToOpenBIS(
+      PostmanCommandLineOptions commandLineParameters) {
+    System.out.format("Please provide password for user '%s':\n", commandLineParameters.user);
```

Comment:
```suggestion
    LOG.info("Please provide password for user '%s':\n", commandLineParameters.user);
```
Why use system out when we have a logger?